redfish-core/lib/account_service.hpp

```diff
@@ -2401,209 +2487,220 @@ inline void requestAccountServiceRoutes(App& app)
         // because of the special handling of ConfigureSelf, it's not able to
         // yet
         .privileges({{"ConfigureUsers"}, {"ConfigureSelf"}})
-        .methods(
-            boost::beast::http::verb::
-                patch)([](const crow::Request& req,
-                          const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-                          const std::string& username) -> void {
-            std::optional<std::string> newUserName;
-            std::optional<std::string> password;
-            std::optional<bool> enabled;
-            std::optional<std::string> roleId;
-            std::optional<bool> locked;
-            std::optional<nlohmann::json> oem;
-            std::optional<std::vector<std::string>> accountType;
-            bool isUserItself = false;
-
-            if (!json_util::readJson(
-                    req, asyncResp->res, "UserName", newUserName, "Password",
-                    password, "RoleId", roleId, "Enabled", enabled, "Locked",
-                    locked, "Oem", oem, "AccountTypes", accountType))
-            {
-                return;
-            }
+        .methods(boost::beast::http::verb::patch)(
+            [](const crow::Request& req,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& username) -> void {
+                std::optional<std::string> newUserName;
+                std::optional<std::string> password;
+                std::optional<bool> enabled;
+                std::optional<std::string> roleId;
+                std::optional<bool> locked;
+                std::optional<nlohmann::json> oem;
+                std::optional<std::vector<std::string>> accountType;
+                bool isUserItself = false;
+
+                if (!json_util::readJson(req, asyncResp->res, "UserName",
```

Comment:
readJsonPatch() instaed of readJson()?
ex: https://grok.openbmc.org/xref/openbmc/bmcweb/redfish-core/lib/account_service.hpp?r=0ec8b83d#1425

Reply:
Good catch!  The code in this branch had not started using the readJsonPatch function, so we will leave it as-is for the fixpack.  We'll change to the code to use readJsonPatch when we rebase for the next release.